builtin/logical/database/path_config_connection.go

```diff
@@ -272,7 +271,16 @@ func (b *databaseBackend) connectionWriteHandler() framework.OperationFunc {
 		}
 
 		resp := &logical.Response{}
-		resp.AddWarning("Read access to this endpoint should be controlled via ACLs as it will return the connection details as is, including passwords, if any.")
+
+		// The is a simple test to to check for passwords in the connection_url paramater. If one exists,
+		// warn the user to use templated url string
+		if connURLRaw, ok := config.ConnectionDetails["connection_url"]; ok {
```

Comment:
Could we add this check to `connectionReadHandler()` and strip out the `connection_url` if a password is found?

Reply:
Addressed in https://github.com/hashicorp/vault/pull/4300/commits/0a329a77a5257ab09db73a106821f1893cbbdf92.